protocol/display: fix Python 3 buffer abstraction

benoit-pierre opened the pull request:
With Python 2, slicing a buffer yield new data, but with Python 3, slicing a memoryview return another memoryview:

```python
> python2 -c 'print buffer(b"12345")[1:3].decode()'
23

> python3 -c 'print(memoryview(b"12345")[1:3].decode())'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'memoryview' object has no attribute 'decode'
```

vasily-v-ryabov commented on Xlib/protocol/display.py:
```diff
@@ -42,12 +42,41 @@
+
+    class bytesview(object):
+
+        def __init__(self, data, offset=None, size=None):
```
Why not just `offset=0` by default? It saves 2 lines below.

vasily-v-ryabov commented on Xlib/protocol/display.py:
```diff
@@ -42,12 +42,41 @@
+            if isinstance(data, bytes):
+                view = memoryview(data)
+            elif isinstance(data, bytesview):
+                view = data._view
```
All static analyzers will produce a warning for this line. Accessing pseudo-private member of another object can be eliminated with a read-only property `view` for the `_view` field (as an option). Or you may make it public for simplicity.

vasily-v-ryabov commented on Xlib/protocol/display.py:
```diff
@@ -42,12 +42,41 @@
+            elif isinstance(data, bytesview):
+                view = data._view
+            else:
+                raise ValueError('unsupported type: %s' % type(data))
```
Maybe `raise TypeError('unsupported type: {}'.format(type(data)))` is better and should work on py2.6+.

benoit-pierre replied:
OK, but we don't support 2.6 anymore. In fact, there's a extraneous `view = memoryview(data)` in the Python 2 code that made me realize that `memoryview` is also supported in Python 2, starting with 2.7, so we could further simplify the code by just using one `BytesView` class for all supported Python versions. What do you think?

vasily-v-ryabov replied:
I remember, we don't support py2.6 (just mentioned it because `.format()` is new in 2.6). I like the idea of `BytesView` class.

vasily-v-ryabov commented on Xlib/protocol/display.py:
```diff
@@ -42,12 +42,41 @@
+
+    def bytesview(data, offset=None, size=None):
+        if not isinstance(data, (bytes, buffer)):
+            raise ValueError('unsupported type: %s' % type(data))
```
The same as above.

benoit-pierre commented:
Thanks for the review, I'll amend.

benoit-pierre commented:
Unfortunately, `memoryview` behaves differently in Python 2 and 3...

```python
> python2 -c 'print bytes(memoryview(b"12345"))'
<memory at 0x2ae8f7a2f9d0>

> python3 -c 'print(bytes(memoryview(b"12345")))'
b'12345'
```

So I kept the 2 different code paths for Python 2 and 3, and addressed your comments.
